ddtrace/tracer/tracer.go

```diff
@@ -624,7 +623,7 @@ func (t *tracer) sample(span *span) {
 	sampler := t.config.sampler
 	if !sampler.Sample(span) {
 		span.context.trace.drop()
-		span.setSamplingPriority(ext.PriorityAutoReject, samplernames.RuleRate)
+		span.context.trace.setSamplingPriority(ext.PriorityAutoReject, SamplingRuleSpan)
```

Comment:
SamplingRuleSpan is the wrong type here this is the value which gets used in _dd.p.dm to mark why we dropped this (although it probably doesn't actually matter here since this trace will be dropped, it's still useful for any possible metrics calculated from this value even if the trace itself isn't ingested)
```suggestion
		span.context.trace.setSamplingPriority(ext.PriorityAutoReject, samplernames.RuleRate)
```